File: src/sdk/platforms/ios/ConsentHub.swift
import Foundation
import UIKit

@objc public class ConsentHub: NSObject {
    public static let shared = ConsentHub()
    private var config: [String: Any]?
    
    public func initialize(clientId: String, apiKey: String) {
        config = [
            "clientId": clientId,
            "apiKey": apiKey,
            "platform": "ios"
        ]
        
        // Initialize SDK
    }
    
    @objc public func showPreferences(viewController: UIViewController) {
        let preferencesVC = ConsentPreferencesViewController()
        let nav = UINavigationController(rootViewController: preferencesVC)
        viewController.present(nav, animated: true)
    }
    
    @objc public func checkConsent(type: String) -> Bool {
        // Implement consent checking
        return false
    }
    
    @objc public func updateConsent(type: String, granted: Bool) {
        // Implement consent updating
    }
    
    @objc public func requestPushNotificationConsent(completion: @escaping (Bool) -> Void) {
        let center = UNUserNotificationCenter.current()
        center.requestAuthorization(options: [.alert, .sound, .badge]) { granted, error in
            completion(granted)
        }
    }
    
    private func getDeviceInfo() -> [String: Any] {
        return [
            "deviceModel": UIDevice.current.model,
            "osVersion": UIDevice.current.systemVersion,
            "appVersion": Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "unknown"
        ]
    }
}

File: src/sdk/platforms/android/ConsentHub.kt
package io.consenthub.sdk

import android.content.Context
import android.content.Intent
import androidx.activity.result.ActivityResultLauncher
import androidx.fragment.app.FragmentActivity

class ConsentHub private constructor(private val context: Context) {
    private var config: Map<String, Any>? = null
    
    companion object {
        @Volatile
        private var instance: ConsentHub? = null
        
        fun getInstance(context: Context) = instance ?: synchronized(this) {
            instance ?: ConsentHub(context).also { instance = it }
        }
    }
    
    fun initialize(clientId: String, apiKey: String) {
        config = mapOf(
            "clientId" to clientId,
            "apiKey" to apiKey,
            "platform" to "android"
        )
        // Initialize SDK
    }
    
    fun showPreferences(activity: FragmentActivity) {
        val intent = Intent(context, ConsentPreferencesActivity::class.java)
        activity.startActivity(intent)
    }
    
    fun checkConsent(type: String): Boolean {
        // Implement consent checking
        return false
    }
    
    fun updateConsent(type: String, granted: Boolean) {
        // Implement consent updating
    }
    
    fun requestPushNotificationConsent(activity: FragmentActivity, launcher: ActivityResultLauncher<String>) {
        launcher.launch(android.Manifest.permission.POST_NOTIFICATIONS)
    }
    
    private fun getDeviceInfo(): Map<String, String> {
        return mapOf(
            "deviceModel" to android.os.Build.MODEL,
            "osVersion" to android.os.Build.VERSION.RELEASE,
            "appVersion" to context.packageManager.getPackageInfo(context.packageName, 0).versionName
        )
    }
}